src/main/java/AudioPlayer.java

```diff
@@ -0,0 +1,240 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.io.IOException;
+
+import models.Song;
+import models.Singer;
+
+/**
+ * This class represent an audio player and it's functions
+ */
+public class AudioPlayer {
+    private List<Song> songs = new ArrayList<>();
+    private Applicaiton applicaiton;
+    private int currentSongIndex;
+    private boolean wasSongPaused;
+    private int songDurationLeft;
+    private static int FIRST_SONG_INDEX = 0;
+
+    public AudioPlayer(Applicaiton app) {
+        applicaiton = app;
+    }
+
+    /**
+     * This methods add a list of songs to the list of songs in the audio player.
+     *
+     * @param songs - this parameter is the list with songs to add
+     */
+    public void addListOfSongs(List<Song> songs) {
+        if (validateSongs(songs)) {
+            this.songs.addAll(songs);
+        }
+    }
+
+    /**
+     * This method check if there are any songs in the list
+     *
+     * @return boolean - the method return true if there is at least one song, if there isn't return false
+     */
+    private boolean validateSongs(List<Song> songs) {
+        return songs != null && songs.size() > 0;
+    }
+
+    /**
+     * This method is used to get AudioPlayer object's characteristics in string format
+     *
+     * @return String - this method return AudioPlayer object's characteristics in string format
+     */
+    @Override
+    public String toString() {
+        StringBuilder audioPlayerInfo = new StringBuilder();
+        audioPlayerInfo.append("\nSongs: ")
+                .append(songs);
+        return audioPlayerInfo.toString();
+    }
+
+    /**
+     * This method executes songs starting from current song position, which is tracked by the class internally. If the songs list
+     * end is reached the playing start from the beginning.
+     */
+    public void playSong() throws IOException, InterruptedException {
+        if (validateSongs(songs)) {
```

Comment:
As discuss in previous comment: if `songs` is never NULL, then this `if (validateSongs(songs)) {` is not needed at all.